internal/hydra/device.go

```diff
@@ -0,0 +1,210 @@
+package hydra
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	hClient "github.com/ory/hydra-client-go/v2"
+)
+
+// We implement the device API logic, as the upstream sdk does not support it.
+// Otherwise we would have to fork the sdk
+// TODO(nsklikas): Remove this once upstream hydra supports the device flow
+
+type DeviceApi interface {
+	AcceptUserCodeRequest(context.Context) ApiAcceptUserCodeRequestRequest
+	AcceptUserCodeRequestExecute(ApiAcceptUserCodeRequestRequest) (*hClient.OAuth2RedirectTo, *http.Response, error)
+}
+
+type DeviceApiService struct {
+	client *hClient.APIClient
```

Comment:
based on the change this would become https://github.com/canonical/identity-platform-login-ui/pull/226#discussion_r1548375587